xbmc/cores/RetroPlayer/RetroPlayer.h

```diff
@@ -128,7 +128,7 @@ class CRetroPlayer : public IPlayer,
   std::unique_ptr<CRetroPlayerInput> m_input;
   std::unique_ptr<IPlayback> m_playback;
   std::unique_ptr<IPlaybackControl> m_playbackControl;
-  std::unique_ptr<CRetroPlayerAutoSave> m_autoSave;
+  mutable std::unique_ptr<CRetroPlayerAutoSave> m_autoSave;
```

Comment:
can you check if we really need to reset this in GetPlayerState, @garbear

Reply:
`GetPlayerState()` is used to create a savestate and return the path, so it shouldn't be made const. If we want the function to be const, then we'll need a new mutable function for savestate creation.

Reply:
okay then. it's not obvious to me why saving state should modify player state, but i guess there are reasons.

Reply:
You have a point. In that regard we'd take my suggestion and add a `CreatePlayerState()` that creates the savestate and stores the path for RP, then `GetPlayerState()` is const and returns the path. Because we use the ApplicationPlayer wrapper we could do (without comments):

```
--- a/xbmc/ApplicationPlayer.cpp
+++ b/xbmc/ApplicationPlayer.cpp
@@ -544,11 +544,14 @@ void CApplicationPlayer::DoAudioWork()
     player->DoAudioWork();
 }
 
 std::string CApplicationPlayer::GetPlayerState() // not const
 {
   std::shared_ptr<IPlayer> player = GetInternal();
   if (player)
-    return player->GetPlayerState();
+  {
+    player->CreatePlayerState(); // {} for all players but RP
+    return player->GetPlayerState(); // const
+  }
   else
     return "";
 }
```

That would make the virtual interface const-correct, which affects all players, and the public interface non-const, which isolates the player interface from this quirk.